src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    [ExportCompletionProvider(nameof(OperatorIndexerCompletionProvider), LanguageNames.CSharp)]
+    [ExtensionOrder(After = nameof(SymbolCompletionProvider))]
+    [Shared]
+    internal class OperatorIndexerCompletionProvider : LSPCompletionProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public OperatorIndexerCompletionProvider()
+        {
+        }
+
+        private const string CompletionHandlerPropertyname = "CompletionHandler";
+        private const string CompletionHandlerConversion = "Conversion";
+        private const string CompletionHandlerIndexer = "Indexer";
+
+        internal override ImmutableHashSet<char> TriggerCharacters => ImmutableHashSet.Create('.');
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '.';
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static ImmutableDictionary<string, string> CreateCompletionHandlerProperty(string operation)
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, string>();
+            builder.Add(CompletionHandlerPropertyname, operation);
+            return builder.ToImmutable();
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
```suggestion
            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
```

and remove null check below.